src/extension.ts

```diff
@@ -228,6 +225,14 @@ export class Extension implements RunHooks {
       this._reusedBrowser,
       this._diagnostics,
       this._treeItemObserver,
+      this._models,
+      this._models.onUpdated(() => {
```

Comment:
Looks like this code operates `this._models` and its content, so it should be a part of `TestModelCollection`. It can probably be handled more efficiently there.